From 80070f1f8f8366a3b24c23a4cf15ac83bdfa41b7 Mon Sep 17 00:00:00 2001
From: Yuanzheng Song <songyuanzheng@huawei.com>
Date: Wed, 2 Nov 2022 06:54:45 +0000
Subject: [PATCH 185/595] [Huawei] Revert "[Huawei] gcc-plugins: change the
 $HOSTCC to the $CC"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186766

--------------------------------

This reverts commit d1193b25155fd7410c855abdf7c126690aa91dcf.

Using $CC to replace $HOSTCC in script/gcc-plugin.sh is a mistake.
As a result, the CONFIG_GCC_PLUGINS can not be set to Y. So fix it
by reverting that patch.

Fixes: d1193b25155f ("[Huawei] gcc-plugins: change the $HOSTCC to the $CC")
Signed-off-by: Yuanzheng Song <songyuanzheng@huawei.com>
---
 scripts/gcc-plugin.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/gcc-plugin.sh b/scripts/gcc-plugin.sh
index 86d8bead6ab2..b79fd0bea838 100755
--- a/scripts/gcc-plugin.sh
+++ b/scripts/gcc-plugin.sh
@@ -8,7 +8,7 @@ srctree=$(dirname "$0")
 gccplugins_dir=$($* -print-file-name=plugin)
 
 # we need a c++ compiler that supports the designated initializer GNU extension
-$CC -c -x c++ -std=gnu++98 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
+$HOSTCC -c -x c++ -std=gnu++98 - -fsyntax-only -I $srctree/gcc-plugins -I $gccplugins_dir/include 2>/dev/null <<EOF
 #include "gcc-common.h"
 class test {
 public:
-- 
2.17.1

